frontend/packages/kubevirt-plugin/integration-tests/tests/vm.actions.scenario.ts

```diff
@@ -0,0 +1,334 @@
+import * as _ from 'lodash';
+import { $, browser, ExpectedConditions as until } from 'protractor';
+import { appHost, testName } from '../../../../integration-tests/protractor.conf';
+import {
+  isLoaded,
+  resourceRowsPresent,
+  textFilter,
+  resourceRows,
+} from '../../../../integration-tests/views/crud.view';
+import { listViewAction, getDetailActionDropdownOptions } from '../views/vm.actions.view';
+import { createNic, networkTypeDropdownId } from '../views/kubevirtDetailView.view';
+import { statusIcon, statusIcons, vmDetailNode } from '../views/virtualMachine.view';
+import {
+  click,
+  removeLeakedResources,
+  deleteResource,
+  deleteResources,
+  createResource,
+  createResources,
+  searchYAML,
+  waitForCount,
+  getDropdownOptions,
+  waitForStringInElement,
+  addLeakableResource,
+  removeLeakableResource,
+} from '../../../console-shared/src/test-utils/utils';
+import { multusNad, hddDisk, networkInterface, getVmManifest } from './utils/mocks';
+import { fillInput, getResourceObject } from './utils/utils';
+import {
+  VM_BOOTUP_TIMEOUT_SECS,
+  VM_STOP_TIMEOUT_SECS,
+  VM_ACTIONS_TIMEOUT_SECS,
+  PAGE_LOAD_TIMEOUT_SECS,
+  TABS,
+  VM_MIGRATION_TIMEOUT_SECS,
+} from './utils/consts';
+import { VirtualMachine } from './models/virtualMachine';
+
+describe('Test VM actions', () => {
+  const leakedResources = new Set<string>();
+  const testVm = getVmManifest('Container', testName);
+
+  afterAll(async () => {
+    removeLeakedResources(leakedResources);
+  });
+
+  describe('Test VM list view kebab actions', () => {
+    const vmName = `vm-list-actions-${testName}`;
+
+    beforeAll(async () => {
+      testVm.metadata.name = vmName;
+      createResource(testVm);
+      addLeakableResource(leakedResources, testVm);
+
+      // Navigate to Virtual Machines page
+      await browser.get(`${appHost}/k8s/ns/${testName}/virtualmachines`);
+      await isLoaded();
+      await fillInput(textFilter, vmName);
+      await resourceRowsPresent();
+    });
+
+    it(
+      'Starts VM',
+      async () => {
+        await listViewAction(vmName)('Start', true);
+        await fillInput(textFilter, vmName);
+        await browser.wait(
+          until.presenceOf(statusIcon(statusIcons.running)),
+          VM_BOOTUP_TIMEOUT_SECS,
+        );
+      },
+      VM_BOOTUP_TIMEOUT_SECS,
+    );
+
+    it(
+      'Restarts VM',
+      async () => {
+        await listViewAction(vmName)('Restart', true);
+        await fillInput(textFilter, vmName);
+        await browser.wait(
+          until.or(
+            until.presenceOf(statusIcon(statusIcons.starting)),
+            until.presenceOf(statusIcon(statusIcons.error)),
+          ),
+        );
+        await browser.wait(
```

Comment:
why is the `running -> error -> running` flow ok? Shouldn't we check against that?

Reply:
I have noticed that sometimes container VM ends up briefly in `error` state during restart and since container VM boots quite quickly and the status is not updated so often, the VM goes from `error` straight to `running`, which broke the automation because it was waiting for `starting` phase.